### PR TITLE
Make pprintast handle new modes

### PR DESCRIPTION
**Pull request by liam923**
Currently, printing the ast of `type a = { x : b @@ portable }` (or any other modality besides `global`) fails. This PR makes this succeed, while still printing global as a legacy modality. For example, `type a = { x : b @@ global portable }` prints as `type a = { x : global_ b @@ portable }`.

### CODE REVIEW

File: ocaml/testsuite/tests/language-extensions/pprintast_unconditional.ml [hunk 1]
```diff
@@ -20,11 +20,13 @@ module Example = struct
 
   let modality_record  = parse module_expr
     "struct \
-      type t = {global_ x : string; global_ y : int} \
```
Comment (freemagma) [inline ocaml/testsuite/tests/language-extensions/pprintast_unconditional.ml:23]: Can you add a test case without a legacy modality in it? All the constructor args in these examples have `global_` on them

File: ocaml/parsing/pprintast.ml [hunk 2]
```diff
@@ -355,10 +365,13 @@ let maybe_modes_type pty ctxt f c =
   | None -> pty ctxt f c
 
 let modalities_type pty ctxt f pca =
-  match pca.pca_modalities with
+  let modalities = Split_modalities.split_list pca.pca_modalities in
+  match modalities.legacy with
   | [] -> pty ctxt f pca.pca_type
```
Comment (freemagma) [inline ocaml/parsing/pprintast.ml:370]: I think this will fail to print the nonlegacy modalities if there are no legacy modalities

Comment (liam923) [reply ocaml/parsing/pprintast.ml:370]: Yep you're right

Comment (liam923) [reply ocaml/parsing/pprintast.ml:370]: Fixed

File: ocaml/testsuite/tests/language-extensions/pprintast_unconditional.ml (hunk 1 above)
Comment (liam923) [reply ocaml/testsuite/tests/language-extensions/pprintast_unconditional.ml:23]: Good point

File: ocaml/parsing/pprintast.ml [hunk 3]
```diff
@@ -355,10 +365,16 @@ let maybe_modes_type pty ctxt f c =
...
-  match pca.pca_modalities with
-  | [] -> pty ctxt f pca.pca_type
-  | m ->
-    pp f "%a %a" legacy_modalities m (pty ctxt) pca.pca_type
+  let modalities = Split_modalities.split_list pca.pca_modalities in
+  let space_if_legacies =
+    match modalities.legacy with
+    | [] -> ""
+    | _ :: _ -> " "
+  in
```
Comment (freemagma) [inline ocaml/parsing/pprintast.ml:373]: there's already a function `optional_legacy_modalities` that should do the space for you. everything else looks good though, so feel free to merge after fixing

**Review by freemagma — APPROVED** [review]

**Pull request merged**